includes/Core/User_Input/User_Input.php

```diff
@@ -0,0 +1,314 @@
+<?php
+/**
+ * Class Google\Site_Kit\Core\User_Input\User_Input
+ *
+ * @package   Google\Site_Kit\Core\User_Input
+ * @copyright 2022 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Core\User_Input;
+
+use Google\Site_Kit\Context;
+use Google\Site_Kit\Core\Authentication\Authentication;
+use Google\Site_Kit\Core\Authentication\User_Input_State;
+use Google\Site_Kit\Core\Permissions\Permissions;
+use Google\Site_Kit\Core\REST_API\REST_Route;
+use Google\Site_Kit\Core\Storage\Options;
+use Google\Site_Kit\Core\Storage\User_Options;
+use WP_Error;
+use WP_REST_Request;
+use WP_REST_Server;
+
+/**
+ * Class managing requests to user input settings endpoint.
+ *
+ * @since n.e.x.t
+ * @access private
+ * @ignore
+ */
+class User_Input {
+
+	/**
+	 * Authentication instance.
+	 *
+	 * @since n.e.x.t
+	 * @var Authentication
+	 */
+	private $authentication;
+
+	/**
+	 * Site_Specific_Questions instance.
+	 *
+	 * @since n.e.x.t
+	 * @var Site_Specific_Questions
+	 */
+	protected $site_specific_questions;
+
+	/**
+	 * User_Specific_Questions instance.
+	 *
+	 * @since n.e.x.t
+	 * @var User_Specific_Questions
+	 */
+	protected $user_specific_questions;
+
+	/**
+	 * User Input questions.
+	 *
+	 * @since n.e.x.t
+	 * @var array|ArrayAccess
+	 */
+	private static $questions = array(
+		'purpose'       => array(
+			'scope' => 'site',
+		),
+		'postFrequency' => array(
+			'scope' => 'user',
+		),
+		'goals'         => array(
+			'scope' => 'user',
+		),
+	);
+
+	/**
+	 * Constructor.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Context        $context         Plugin context.
+	 * @param Authentication $authentication  Optional. Authentication instance. Default a new instance.
+	 * @param Options        $options         Optional. Options instance. Default a new instance.
+	 * @param User_Options   $user_options    Optional. User_Options instance. Default a new instance.
+	 */
+	public function __construct(
+		Context $context,
+		Authentication $authentication = null,
```

Comment:
We probably shouldn't need `Authentication here`. See below.

Reply:
Removed, thank you!